overlays/name-and-version.nix

```diff
@@ -0,0 +1,20 @@
+final: prev:
+let
+  inherit (prev) lib;
+  inherit (import ../lib { inherit lib; }) checkMkDerivationFor;
+
+  checkDerivation = drvArgs: drv:
+    lib.singleton {
+      name = "name-and-version";
+      cond = drvArgs ? name && drvArgs ? version;
```

Comment:
Maybe we should only show this when `pname` is not present or `drvArgs` or when `drvArgs.name == "${drvArgs.pname}-${drvArgs.version}"`.

Reply:
WIth first case -- agree. In case when all three -- name, pname and version are present -- I do not know. It would very, very strange, but I guess developer who does it has some idea what he wants.

Reply:
Sometimes `name = "${pname}-${variant}-${version}"` is used, which is IMO legit. `name = "${pname}-${version}"` is most likely just an artefact from the time before `stdenv.mkDerivation` accepted `pname`. But maybe the second case is not frequent enough to warrant an exception from the exception for the first case.

Reply:
Okay, pushed new version. If name and version are passed, and pname is not -- warning is raised.